### DISCUSSION TITLE
Bad UX with scrolling menus on Chrome/Safari on MacOS

### DISCUSSION THREAD

**Discussion by dimitrov-adrian** [directus/directus#14653] [Feature Requests]
## The problem

With Chrome and Safari there is a an annoying issue with menus that have scrolling items, initially when the menu is open the beyond the fold items are invisible and there is no sign they appears. This problem affects (AFAIK) MacOS when enabled different option than "Always" under "Scrollbar bar show" settings. So ending with something like

![image](https://user-images.githubusercontent.com/1009639/180984767-e43cc95f-666e-4dae-ab6f-4c9553655948.png)

## Approaches

As a possible solutions I have created a two different approaches as simple fix for that.

1. A GTK/Gnome alike approach which adds a dashed line at the bottom when there are items beyond the fold

https://user-images.githubusercontent.com/1009639/180985286-4759efda-83aa-4af8-8079-b0748e507ebd.mov

2. Using css scrollbar styling to force the always on scrollbars on the menus (first part is with Chrome, second - Safari)

https://user-images.githubusercontent.com/1009639/180985353-169d4cf3-704b-47d8-aaa0-c350ff7abb4f.mov

I can make a PR if align with some of the approaches that I have tested

**Comment by azrikahar** [comment]
Definitely a valid concern, which is the same as #14468.

For the first approach, I wonder if we can do the gradient and/or arrow similar to how Youtube does it (but vertically):

![chrome_ap2GBbyR6H](https://user-images.githubusercontent.com/42867097/180988505-df45c137-8cd4-4405-8c4e-99305cd3fe67.gif)

Just throwing ideas here 😄

As for second approach, that does feel like the more straight forward solution 👍 let's see what others think of it.

**Reply by benhaynes** [reply]
I like the idea of using a gradient/shadow to show that there is more content to scroll to!

**Reply by rijkvanzanten** [reply]
Tricky to do efficiently within the context of those popovers though 🤔 A good in-between approach would be to set the min/max-height of the popovers to a value that doesn't perfectly align with `x` items in the v-list. If you can see half of the next list item, that's already a huge hint that there's more content to scroll to. (We wouldn't want to use `overflow: scroll`, as that would render the scrollbars at all times, even if there's nothing to scroll to)

**Reply by dimitrov-adrian** [reply]
I agree that it's quite tricky to do with the fading, as it possible to break this magic number because of the dividers, so it might need of a calculation on popup open and set the height after that, but it's doable solution as well.

In my second approach I am not using `overflow: scroll` but styling a scrollbar thumbs to feel like always on, and it behave that only if there is content to scroll, since the `overflow: scroll` do not show scrollbar at all on macOS when "Show scroll bar" is set to "When scrolling"

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
